### DISCUSSION TITLE
Custom Field Validation

### DISCUSSION THREAD

**Discussion by reaboivictor** [statamic/cms#6829] [Q&A]
Hi, let's suppose we have this Location custom field. I am validating the location key only (everything works as expected).

What I want to do is validate all the keys as required. Can someone give a hint if there is a way of doing it?

Thanks in advance!

```
<?php

namespace App\Fieldtypes;

use Statamic\Fields\Fieldtype;

class Location extends Fieldtype
{
    public function defaultValue()
    {
        return [
            'location' => '',
            'long' => '',
            'lat' => '',
        ];
    }
    
    public function preProcessValidatable($value)
    {
        return $value['location'];
    }
}

```

**Comment by edalzell** [comment]
I don't quite understand sorry, what other fields do you want to validate?

**Reply by reaboivictor** [reply]
I want to validate all array keys: location, long and lat
